_specification-v1/imageset.md

```diff
@@ -0,0 +1,109 @@
+description: A specification which describes the required structure of an imageset.json manifest file.
+cta: Read the imageset manifest spec
+
+# Navigation config
+nav_display: true
+nav_label: Imageset
+nav_heading_selector: h1, h2, h3
+---
+
+# {{page.title}}
+
+`imageset.json` is a <a href="https://www.json.org/" class="o-typography-link--external"><abbr title="JavaScript Object Notation">JSON</abbr></a> format file that is responsible for describing various aspects of an Origami Imageset.
+
+## Properties
+
+### sourceDirectory
+
+<table class="o-manifest__table o-table o-table--compact o-table--row-headings o-table--vertical-lines o-table--horizontal-lines" data-o-component="o-table">
+	<tr>
+		<th scope="row" role="rowheader">Type</th>
+		<td><code>String</code></td>
+	</tr>
+	<tr>
+		<th scope="row" role="rowheader">Required</th>
+		<td><code>true</code></td>
+	</tr>
+</table>
+
+**Must** be a the relative path to the folder which contains the images within the imageset.
+<pre><code class="o-syntax-highlight--json">{
+	"sourceDirectory": "src"
+}</code></pre>
+
+### scheme
+
+<table class="o-manifest__table o-table o-table--compact o-table--row-headings o-table--vertical-lines o-table--horizontal-lines" data-o-component="o-table">
+	<tr>
+		<th scope="row" role="rowheader">Type</th>
+		<td><code>String</code></td>
+	</tr>
+	<tr>
+		<th scope="row" role="rowheader">Required</th>
+		<td><code>true</code></td>
+	</tr>
+</table>
+
+Defines the custom url scheme for images in this imageset when requested via the [Origami Image Service](https://www.ft.com/__origami/service/image/v2/), this **must** be a unique name across all the imagesets:
+<pre><code class="o-syntax-highlight--json">{
+	"scheme": "ftanimal"
+}</code></pre>
+
+### images
+
+<table class="o-manifest__table o-table o-table--compact o-table--row-headings o-table--vertical-lines o-table--horizontal-lines" data-o-component="o-table">
+	<tr>
+		<th scope="row" role="rowheader">Type</th>
+		<td><code>Array</code></td>
+	</tr>
+	<tr>
+		<th scope="row" role="rowheader">Required</th>
+		<td><code>true</code></td>
+	</tr>
+</table>
+
+A list of image objects for each image in the imageset.
+Each object in the list accepts the following properties:
+
+**required**:
+- `name`: type `String`. The name which will be used together with the <a href="#scheme">scheme</a> to requested the image via the [Origami Image Service](https://www.ft.com/__origami/service/image/v2/)
+- `extension`: type `String`. The extension for the file type of the image
+- `path`: type `String`. The relative path to the image within the imageset
+- `previousHash`: type `String` or `null` or `undefined`. The previous hash of the image, this will be `null` or `undefined` if the image has never been updated
+- `hash`: type `String`. The hash of the image
+- `url`: type `String`. The production [Origami Image Service](https://www.ft.com/__origami/service/image/v2/) url which points to the image
```

Comment:
```suggestion
- `url`: type `String`. The production [Origami Image Service](https://www.ft.com/__origami/service/image/v2/) url which points to the image

**optional**:
- `deprecated`: type `String`. A deprecation message for the image to indicate the image should no longer be used. The existence of the deprecated key with an empty string indicates the image is deprecated but has no specific deprecation message.
```

Reply:
☝️introduces a new key, as in the proposal

Reply:
Hmm, now that I see the proposal inside an imageset manifest I think I see a problem because this whole section (the `images` array) is currently created and updated by the tooling https://github.com/Financial-Times/origami-image-set-tools/blob/master/lib/origami-image-set-tools.js#L85-L93

I think we should maybe have another section for the deprecated property to avoid have a single property in the image objects which is done manually and all the others and automated. What do you think @financial-times/origami-core?

Reply:
That's a good point! Do you have any ideas for another section? I'm thinking `deprecatedImages: {imagename: deprecationMessage}` but it feels clunky. `oist` is already working with the current manifest, maybe we should prioritise the human editor and update it to pass through the `deprecated` key?

Reply:
Yeah, we can have origami-image-set-tools ignore any extra properties added to the image objects 👍 Let's do that :-)